cdk/visit/migration_scripts/migrate_data_to_2_tables.py

```diff
@@ -0,0 +1,162 @@
+"""
+This script will migrate data from the original table to the two
+new tables.
+"""
+from typing import Tuple, List
+import boto3
+import os
+import datetime
+
+
+def generate_role_arn() -> str:
+    """
+    Generate a role arn for the migration script.
+
+    Returns:
+        A role arn.
+    """
+    return 'arn:aws:iam::{}:role/{}'.format(os.environ['AWS_ACCOUNT_ID'], 'data_migration_role')
+
+
+def get_table(table_name: str) -> boto3.resources.base.ServiceResource:
+    """
+    Get a table resource from the boto3 resource manager.
+
+    Args:
+        table_name: The name of the table to get.
+
+    Returns:
+        A table resource.
+    """
+    dynamodb = boto3.resource('dynamodb', region_name='us-east-1')
+    table = dynamodb.Table(table_name)
+    return table
+
+
+def get_all_data(table: boto3.resources.base.ServiceResource) -> list:
+    """
+    Get all data from a table.
+
+    Args:
+        table: The table to get data from.
+
+    Returns:
+        A list of all data from the table.
+    """
+    response = table.scan()
+    return response['Items']
+
+
+def process_grad_date(grad_date: str) -> Tuple[str, int]:
+    """
+    Infers the graduation semester and year from grad_date
+
+    grad_date: str
+        The graduation date in the format 'YYYY-MM-DD'
+
+    Returns:
+        A tuple of the semester and year.
+    """
+    year = grad_date[:4]
+    month = grad_date[5:7]
+    if month in ['04', '05', '06']:
+        semester = 'Spring'
+    elif month in ['07', '08', '09']:
+        semester = 'Summer'
+    elif month in ['11', '12', '01']:
+        semester = 'Fall'
+    else:
+        semester = "None"
+
+    return semester, int(year)
+
+
+def process_timestamp(timestamp: str) -> str:
```

Comment:
Should this return some sort of float rather than a string since its represented as a Num in DDB?

Reply:
The marshaled data entry toward the end of the migration script, requires that the RHS of the field be a string for the number type, "N" (https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_AttributeValue.html). However ideally the function would return a type intuitive to someone reading the code expecting a timestamp (float or int), and not have any coupling to the storage type requirements later in the code.